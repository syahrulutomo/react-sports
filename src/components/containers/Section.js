import React from 'react';
import PropTypes from 'prop-types';

export function Section({ children, style }) {
  return <section className="section--default" style={style}>{children}</section>;
}

Section.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.func,
    PropTypes.object,
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
  style: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
};
