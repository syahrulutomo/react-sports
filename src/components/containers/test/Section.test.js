import React from 'react';
import {cleanup} from '@testing-library/react';
import { Section } from '../Section'
import ShallowRenderer from 'react-test-renderer/shallow';

afterEach(cleanup)

const children = (
  <div></div>
)

it('renders correctly Section component', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Section children={children}/>);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});