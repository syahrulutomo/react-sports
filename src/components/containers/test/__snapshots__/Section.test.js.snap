// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Section component 1`] = `
<section
  className="section--default"
>
  <div />
</section>
`;
